src/System.Memory/src/System/SpanHelpers.BinarySearch.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class SpanHelpers
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int BinarySearch<T, TComparable>(
+            this ReadOnlySpan<T> span, TComparable comparable)
+            where TComparable : IComparable<T>
+        {
+            if (comparable == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.comparable);
+            // TODO: Make `ref readonly`/`in` when Unsafe.Add(ReadOnly) supports it
```

Comment:
@ahsonkhan, @nietras , do we have workitems filed for these Unsafe APIs and to fix the TODOs?

Reply:
Just found https://github.com/dotnet/corefx/issues/25814.

Reply:
I should probably just remove the TODOs since the `Unsafe` fix was to "cast" away the `readonly` and then there is no reason as such to change the impl, if `DangerousGetPinnableReference()` will still return a `ref` for `ReadOnlySpan<T>` otherwise the `readonly` simply has to be cast away. This will show up if any such change comes to `ReadOnlySpan<T>` so deleting the TODOs should be fine.